### ISSUE TITLE
maven-bloop doesn't include directories with generated sources

### ISSUE THREAD

**Issue by nigredo-tori** [scalacenter/bloop#1573]
Consider this maven project: [maven-example.zip](https://github.com/scalacenter/bloop/files/7231391/maven-example.zip). It includes a protobuf model for which Java code is generated, which is later compiled together with the unmanaged sources. Documentation instructs us to export it as follows:

```sh
$ mvn ch.epfl.scala:maven-bloop_2.13:1.4.9:bloopInstall
```
(actually, documentation says 2.12, but that artifact does not exist). However, when exported this way, we only have one source directory in the generated bloop project:
```json
        "sources": [
            "/home/nigredo/dev/tmp/foo/src/main/scala"
        ],
```
whereas the correct output would look like this:
```json
        "sources": [
            "/home/nigredo/dev/tmp/foo/src/main/scala",
            "/home/nigredo/dev/tmp/foo/target/generated-sources/protobuf/java"
        ],
```

This is because both `bloopInstall` and Protobuf code generation are bound to the same Maven phase, `generate-sources`. Thus there is no dependency between these two goals, and Protobuf code generation is not run. Since one of the things the code generation goal does is to add a compile source directory to the project description, and that does not happen, maven-bloop has no way to know about that directory.

One possible solution would be to bind `bloopInstall` to the `process-sources` phase, so that it is guaranteed to run after code generation. I'm not sure what this would break.

As a workaround, it's possible to manually micromanage the goals by exporting the project this way:

```sh
$ mvn generate-sources ch.epfl.scala:maven-bloop_2.13:1.4.9:bloopInstall
```
This gives us the expected `sources`.

**Comment by tgodzik** [comment]
Thanks for reporting! We should be able to switch to `process-sources`, let me try to experiment with that.

**Comment by tgodzik** [comment]
I tested it out and it seems there is not other way than adding `generate-sources`.

I updated the [docs](https://github.com/scalacenter/bloop/pull/1575) and will add the change to Metals also.

**Comment by tgodzik** [comment]
I updated the docs for Bloop and changed the way Metals runs the Maven command. Otherwise, I don't see any better workaround, because when Bloop is run as an outside plugin it will not invoke other phases.

Some power users might want to add Bloop to `pom.xml` and bind it to the `process-sources` phase, but we don't want to force it on people.

Closing it for now, unless anyone comes up with a better solution. Thanks again @nigredo-tori for raising this!
